CMakeLists.txt

```diff
@@ -27,6 +28,11 @@ add_library(bfs
   src/States/Algorithms/BFS/Node.cpp
 )
 
+add_library(dfs
+  src/States/Algorithms/DFS/DFS.cpp
+  #src/States/Algorithms/DFS/Node.cpp
```

Comment:
I assume the algorithm is just using Node class from BFS, which is not the appropriate way. I think it's better to change the class name definition from `class Node` to `class DFS_Node` so that there will be no naming conflicts during the linking. You can also change the name of the class file from _Node.cpp_ to _DFS_Node.cpp_.